File: frontend/gameapp-frontend/src/domain/entities/scenes/final-scene.ts
import { Scene } from "./scene";
import { SceneName } from "../../value-objects/scenes/scene-name";
import { SceneDescription } from "../../value-objects/scenes/scene-description";
import { Biome } from "../../enumerates/biome";

export class FinalScene extends Scene {
    constructor(name: SceneName, description: SceneDescription, biome: Biome, id?: string) {
        super(name, description, biome, id);
    }

    setSceneName(newName: SceneName) {
        return new FinalScene(newName, this._description, this._biome, this._id);
    }
    setSceneDescription(newDescription: SceneDescription) {
        return new FinalScene(this._name, newDescription, this._biome, this._id);
    }
    setBiome(newBiome: Biome) {
        return new FinalScene(this._name, this._description, newBiome, this._id);
    }

    toString(): string {
        return `${this._name} Final Scene(${this._id}): Description=${this._description}, Biome=${this._biome}`;
    }
}

File: backend/GameApp.Domain/Entities/Scenes/FinalScene.cs
using GameApp.Domain.Enumerates;
using GameApp.Domain.ValueObjects.Scenes;

namespace GameApp.Domain.Entities.Scenes;

// Class for final scene of a game
public class FinalScene : Scene
{

    // Constructor
    public FinalScene(SceneName name, SceneDescription description, Biome biome)
    : base(name, description, biome)
    {
    }

    // Restore Constructor
    public FinalScene(Guid id, SceneName name, SceneDescription description, Biome biome)
    : base(id, name, description, biome)
    {
    }

    // Setters 
    public FinalScene SetSceneName(SceneName newName) =>
    new FinalScene(GetGuid(), newName, GetDescription(), GetBiome());

    public FinalScene SetSceneDescription(SceneDescription newDescription) =>
    new FinalScene(GetGuid(), GetName(), newDescription, GetBiome());

    public FinalScene SetBiome(Biome newBiome) =>
    new FinalScene(GetGuid(), GetName(), GetDescription(), newBiome);

    // To string
    public override string ToString()
    {
        return $"{GetName()} Final Scene({GetGuid()}): " +
               $"Description={GetDescription()}, Biome={GetBiome()}, ";
    }

}